raft/src/test/java/org/apache/kafka/snapshot/SnapshotWriterReaderTest.java

```diff
@@ -151,6 +194,57 @@ public void testAppendToFrozenSnapshot() throws Exception {
         );
     }
 
+    private int validateDelimiters(
+        RawSnapshotReader snapshot,
+        long lastContainedLogTime
+    ) {
+        assertNotEquals(0, snapshot.sizeInBytes());
+
+        int countRecords = 0;
+
+        Iterator<RecordBatch> recordBatches = Utils.covariantCast(snapshot.records().batchIterator());
+
+        assertEquals(Boolean.TRUE, recordBatches.hasNext());
+        RecordBatch batch = recordBatches.next();
+
+        Iterator<Record> records = batch.streamingIterator(new GrowableBufferSupplier());
+
+        // Verify existence of the header record
+        assertEquals(Boolean.TRUE, batch.isControlBatch());
+        assertEquals(Boolean.TRUE, records.hasNext());
+        Record record = records.next();
+        countRecords += 1;
+
+        SnapshotHeaderRecord headerRecord = ControlRecordUtils.deserializedSnapshotHeaderRecord(record);
+        assertEquals(headerRecord.version(), ControlRecordUtils.SNAPSHOT_HEADER_HIGHEST_VERSION);
+        assertEquals(headerRecord.lastContainedLogTime(), lastContainedLogTime);
+
+        assertEquals(Boolean.FALSE, records.hasNext());
+
+        // Loop over remaining records
+        while (recordBatches.hasNext()) {
+            batch = recordBatches.next();
+            records = batch.streamingIterator(new GrowableBufferSupplier());
+
+            while (records.hasNext()) {
+                countRecords += 1;
+                record = records.next();
+            }
+        }
+
+        // Verify existence of the footer record
+        assertEquals(Boolean.TRUE, batch.isControlBatch());
+
+        SnapshotFooterRecord footerRecord = ControlRecordUtils.deserializedSnapshotFooterRecord(record);
+        assertEquals(footerRecord.version(), ControlRecordUtils.SNAPSHOT_HEADER_HIGHEST_VERSION);
+
+        // Verify there is nothing past the footer
+        assertEquals(Boolean.FALSE, records.hasNext());
+        assertEquals(Boolean.FALSE, recordBatches.hasNext());
```

Comment:
These two checks will always pass because of the `while` loops above.

Reply:
Yep. Just an old habit of asserting things to aid readability. Will remove it here.